pyccel/ast/operators.py

```diff
@@ -393,11 +394,11 @@ def _calculate_dtype(cls, *args):
             return cls._handle_str_type(strs)
             assert len(integers + floats + complexes) == 0
```

Comment:
Will it ever pass to this line?

Reply:
@ohachim good catch. I suspect that's my error originally.
@Pinkyboi would you mind changing the order of these lines

Reply:
@EmilyBourne wouldn't it be better to leave just the `TypeError` raised by  `_handle_str_type` ? in my opinion it's more descriptive of the error than just an `assert` although the error message `"unsupported operand type(s) for /: 'str' and 'str'"` could maybe be changed.

Reply:
Ah if there is a `TypeError` then yes that is much better. In that case the `assert` can indeed be removed.
> `"unsupported operand type(s) for /: 'str' and 'str'"` could maybe be changed.

This doesn't seem like it's reporting the same error though, so we might need to add an additional `TypeError`. This is talking about doing `'ha'/'out'` which doesn't mean anything in python. Some of the other operators are supported but I think in that case they have their own classes which are not operators (e.g `Concatenate`).